src/app/context/btcWallet/BtcWallet.tsx

```diff
@@ -0,0 +1,56 @@
+import { networks } from "bitcoinjs-lib";
+import React, { createContext, ReactNode, useContext, useState } from "react";
+
+import { WalletProvider } from "@/utils/wallet/wallet_provider";
+interface BtcWalletContextType {
+  publicKeyNoCoord: string;
+  address: string;
+  btcWallet: WalletProvider | undefined;
+  btcWalletNetwork: networks.Network | undefined;
+  setPublicKeyNoCoord: (key: string) => void;
+  setAddress: (address: string) => void;
```

Comment:
Why do we need the `setPublicKeyNoCoord`, `setAddress` and `setBtcWalletNetwork` methods? These data should not be different than what `btcWallet` returns.